transaction-view/src/transaction_data.rs

```diff
@@ -0,0 +1,12 @@
+/// Trait for accessing transaction data from an abstract byte container.
+pub trait TransactionData {
```

Comment:
Other implementations will be added as it becomes needed.
Can implement this with `Packet` and have the `TransactionView` actually take ownership of the data.
Can also do impl with some `Arc<PacketBatch> + index` to access a specific packet without copying it.